lib/Elastica/Search.php

```diff
@@ -462,7 +463,7 @@ public function search($query = '', $options = null)
             $params
         );
 
-        return ResultSet::create($response, $query);
+        return $this->_client->getResultSetBuilder()->buildResultSet($response, $query);
```

Comment:
I would prefer to inject the ResultSetBuilder directly into the Search object, but I think given the way Elastica is structured that developers will be creating these objects directly and we dont want to introduce such a major change..